src/borg/repository.py

```diff
@@ -321,7 +325,10 @@ def get_index_transaction_id(self):
 
     def check_transaction(self):
         index_transaction_id = self.get_index_transaction_id()
-        segments_transaction_id = self.io.get_segments_transaction_id()
+        try:
+            segments_transaction_id = self.io.get_segments_transaction_id()
+        except LoggedIO.AtticSegment:
+            raise self.AtticRepository(self.path)
```

Comment:
well, the point was to have nice traceback information.
whether we then show it or not is decided elsewhere.